paperqa/settings.py

```diff
@@ -196,6 +196,17 @@ class ParsingSettings(BaseModel):
         ),
     )
     chunking_algorithm: ChunkingOptions = ChunkingOptions.SIMPLE_OVERLAP
+    doc_filters: list[dict] | None = Field(
```

Comment:
More of an aside, but for stuff like this I have seen people use Django REST: https://github.com/encode/django-rest-framework/blob/3.15.1/rest_framework/filters.py#L68-L73

It basically lets you query the DB for any model using URL filters, as opposed to us DIY'ing it behind FastAPI endpoints

More just sharing a way one can do this without DIY'ing filtering logic

Reply:
Agree - would be nice to use something built in, but those look like they're slightly different meanings. We have a lot of optional/possibly missing params. But thanks for sharing - interesting to see!